auth/handler.go

```diff
@@ -221,7 +223,20 @@ func (h *AuthenticationHandler) Authenticate(username string, password string) (
 		}
 	}
 
-	body, _ := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode == 202 {
+		h.logger.Infof("user %s has given correct credentials, but additional authentication factor is required", username)
+		body, _ := io.ReadAll(resp.Body)
+		var unmarshalledBody map[string]interface{}
+
+		if err := json.Unmarshal(body, &unmarshalledBody); err != nil {
```

Comment:
Also, is the response guaranteed to always contain JSON? Maybe add a check that asserts the `Content-Type` in the response?